astropy/units/core.py

```diff
@@ -2072,7 +2073,7 @@ def add_unit(unit, power, scale):
         new_parts.sort(key=lambda x: (-x[1], getattr(x[0], 'name', '')))
 
         self._bases = [x[0] for x in new_parts]
-        self._powers = [validate_power(x[1]) for x in new_parts]
```

Comment:
By this time, the power has already been validated, and it cannot change in a way that makes it invalid.
(but scale can change, hence that check remains in place)